File: src/main/java/lol/pbu/ZcmiCommand.java
package lol.pbu;

import io.micronaut.configuration.picocli.PicocliRunner;

import io.micronaut.context.annotation.Value;
import picocli.CommandLine;
import picocli.CommandLine.Command;
import picocli.jansi.graalvm.AnsiConsole;

@Command(name = "zcmi", description = "...", versionProvider = ZcmiCommand.ZcmiVersionProvider.class, mixinStandardHelpOptions = true)
public class ZcmiCommand extends BaseCommand implements Runnable {

    @Generated
    public static void main(String[] args) {
        try (AnsiConsole ignored = AnsiConsole.windowsInstall()) {
            PicocliRunner.run(ZcmiCommand.class, args);
        }
    }

    public void run() {
        err("No Command Given");
    }

    static class ZcmiVersionProvider implements CommandLine.IVersionProvider {
        @Value("${micronaut.application.version}")
        String version;

        @Override
        public String[] getVersion() {
            return new String[]{version};
        }
    }
}

File: src/main/kotlin/lol/pbu/ZcmiCommand.kt
package lol.pbu

import io.micronaut.configuration.picocli.PicocliRunner
import io.micronaut.context.annotation.Value
import lol.pbu.ZcmiCommand.ZcmiVersionProvider
import picocli.CommandLine
import picocli.CommandLine.Command
import picocli.jansi.graalvm.AnsiConsole

@Command(
    subcommands = [HelpCenterCommand::class],
    mixinStandardHelpOptions = true,
    name = "Zendesk CLI Management Interface",
    versionProvider = ZcmiVersionProvider::class,
    description = ["Work seamlessly with Zendesk from the command line."]
)
class ZcmiCommand : BaseCommand(), Runnable {
    class ZcmiVersionProvider : CommandLine.IVersionProvider {
        @Value("\${micronaut.application.version}")
        var zcmiVersionProvider: String? = null
        override fun getVersion(): Array<out String?> {
            return arrayOf(zcmiVersionProvider); }
    }

    override fun run() {
        err("No command specified.")
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            AnsiConsole.windowsInstall().use { _ -> PicocliRunner.run(ZcmiCommand::class.java, *args) }
        }
    }
}